src/vit-testing/vitup/src/builders/helpers/qr.rs

```diff
@@ -25,9 +26,17 @@ pub fn generate_qr_and_hashes<P: AsRef<Path>>(
     parameters: &Config,
     folder: P,
 ) -> Result<(), Error> {
+    let span = span!(Level::TRACE, "qr code generation");
+    let _enter = span.enter();
+
     let total = wallets.len();
     let folder = folder.as_ref();
 
+    info!(
+        "generating {} valid qr codes and payloads...",
+        wallets.len()
+    );
```

Comment:
I don't know if this is useful here or somewhere else so I'm just sharing this: you can include fields in the span and events as described here https://docs.rs/tracing/latest/tracing/#recording-fields.

Reply:
Good point @FelipeRosa . Moving it to span